src/Metalama.Extensions.DependencyInjection/Implementation/DefaultDependencyInjectionStrategy.cs

```diff
@@ -85,11 +84,22 @@ public virtual bool TryIntroduceDependency( IAspectBuilder<INamedType> builder,
             return true;
         }
 
+        SuppressUnusedWarnings( builder, fieldOrProperty );
+
         var pullStrategy = this.GetPullStrategy( fieldOrProperty );
 
         return this.TryPullDependency( builder, fieldOrProperty, pullStrategy );
     }
 
+    /// <summary>
+    /// Suppress the warning CS0169 ("The private field is never used.") on a member that is being introduced.
+    /// This is primarily useful for design-time.
+    /// </summary>
+    protected static void SuppressUnusedWarnings( IAspectBuilder builder, IFieldOrProperty introducedMember )
```

Comment:
Move these new methods to a public helper type `SuppressionHelper` so it can be used by other implementations without relying on inheritance.